wrappers/csharp/Managed.cs

```diff
@@ -910,6 +948,29 @@ public static string DecryptWithPasswordAsUtf8String(string b64data, string pass
             }
         }
 
+        public static byte[][] GenerateSharedKey(int nbShares, int threshold, int secretLength, byte[][] shares)
+        {
+            try
+            {
+                if (shares != null)
+                {
+                    IntPtr[] pointers = InitializeArray(ref shares, nbShares, secretLength);
+                    long result = Native.GenerateSharedKeyNative((UIntPtr)nbShares, (UIntPtr)threshold, (UIntPtr)secretLength, pointers);
+                }
+
+                return shares;
+            }
+            catch (DevolutionsCryptoException ex)
+            {
+                if (ex.NativeError != null)
+                {
+                    Utils.HandleError((long)ex.NativeError);
+                }
+            }
```

Comment:
1. You should free the pointers after you're done with it.
2. The native errors aren't properly handled